pkg/controller/controller.go

```diff
@@ -538,6 +559,25 @@ func (s *Service) DeleteVolume(ctx context.Context, req *csi.DeleteVolumeRequest
 				"unable to delete volume -- %d snapshots based on this volume still exist.", len(listSnaps))
 		}
 
+		// Check if volume has metro session and end it
+		volume, err := arr.GetClient().GetVolume(ctx, id)
+		if err != nil {
+			if apiError, ok := err.(gopowerstore.APIError); ok && apiError.NotFound() {
+				log.Infof("Volume %s not found, it may have been deleted.", id)
+				return &csi.DeleteVolumeResponse{}, nil
+			}
```

Comment:
Would a warning be better? You don't have to change the code either way.

Reply:
I noticed repeated delete calls to the driver whether it is metro or regular volume. In those cases, 'info' seemed fine.